### PR TITLE
[IOTDB-151]modify antlr3 to support timeseries path in the number format

### PR DESCRIPTION
**Pull request by LeiRui**
To support series path in number format, we have the following logic in TSParser.g already:
```
nodeName
    : identifier
    | STAR
    ;

identifier
    :
    Identifier | integer
    ;
```

Then the main challenge left here is to teach antlr3 to recognize `integer` in the series path, which is easy to be confused with and recognized as `Float`, `DATETIME` partly because of the "Longest match wins first" rule of Antlr3.

For example,
 
|  integer   |                           float                            |                      datetime                      | seriesPath in the number format |
|------------|------------------------------------------------------------|----------------------------------------------------|------------------------------|
| +12 / -12 / 12 | +12.233 / -12.233 / +12.3332e-1 / -12.3332e+2  / 12.3332 / 12.3332e-1 | 2017-11-1T00:15:00+08:00 / 2017.11.1 00:15:00+08:00 | 8955.1227.223        |

The solution is to do a more elaborate division and some are directly recognized in TSLexer and some are later combined in TSParser: 

In TSLexer, we have:
- PositiveInteger, e.g., `+12`
- NegativeInteger, e.g., `-12`
- PositiveFloat, e.g., `+12.233`, `+12.3332e-1 `
- NegativeFloat, e.g., `-12.233`, `-12.3332e-1 `
NOTE that the existence of '+' and '-' help ease the match rule. For those without signs, things get tricky as follows.
- UnsignedInteger, e.g., `12`
- DATETIME, e.g., `2017.11.1 00:15:00+08:00`. 
NOTE that here an critical trick is used (see code). If the first lexer rule of DATETIME is not matched, then UnsignedInteger is recognized instead.
- DoubleInScientificNotationSuffix, e.g., `.E+10`. 
NOTE that here the same critical trick is used (see code). If the first rule is not matched, then DOT is recognized instead.

In TSParser, we have:
- nonNegativeInteger (which is needed by LIMIT&SLIMIT clause)
- integer
- floatValue
NOTE that "float" shoud not be used as a name as it is reserved.
NOTE that the old FLOAT in the TSLexer is removed and floatValue in TSParser is used instead. It is because we want to support series path in the number format, **which can not be separated from the float number in the lexer level**.

As a result, 
- integer and float with signs (+ or -) will each be recognized as a single token. For example, `+12`, `-12`, `+2.3`, `-2.3e10`
- float without signs will be recognized as a set led by `TOK_FLOAT_COMB`. For example, 
  - `2.3` -> `TOK_FLOAT_COMB`, `2`, `.`, `3` (corresponding to `UnsignedInteger DOT UnsignedInteger`)
  - `2.3e10` -> `TOK_FLOAT_COMB`, `2`, `.3e10` (corresponding to `UnsignedInteger DoubleInScientificNotationSuffix`)

Lastly, some notes for user guide:
- The first nodeName of a seriesPath still must be `root`, e.g., root.123.456.789
- The other nodeNames will support three formats:
  - Identifier. i.e., string starts with letter('a'..'z' | 'A'..'Z') or  '_'
  - integer
  - STAR (i.e., *)

### CODE REVIEW

**Comment by LeiRui** [comment]
I prefer `rebase and merge` of this pr.

File: server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TSLexer.g [hunk 1]
```diff
@@ -146,25 +146,51 @@ StringLiteral
...
 
-NonNegativeInteger
+PositiveInteger
+    :
+    '+' Digit+
+    ;
+
+NegativeFloat
+    :
+    '-' Digit+ DOT Digit+ (('e' | 'E') ('+'|'-')? Digit+)?
```
[the remark this answers is not in the document]
Comment (jixuan1989) [reply server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TSLexer.g]: I tried Java:
```
  System.out.println(Float.valueOf(".3"));
  System.out.println(Float.valueOf("3."));
```
Java works in these cases.

I am not sure whether we need to support that...

[the remark this answers is not in the document]
Comment (jixuan1989) [reply server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TSLexer.g]: How about support the cases like ".3" or "3." in new PR if we want..
Then we can quickly support number format in time series path..

[the remark this answers is not in the document]
Comment (LeiRui) [reply server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TSLexer.g]: I will try quick fix. If it turns out to be nontrivial, we can do as @jixuan1989 suggests.

[the remark this answers is not in the document]
Comment (LeiRui) [reply server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TSLexer.g]: Now support float format exmaples:
- `+2.2`
- `+2.`
- `+.2`
- `+2.3e10`
- `+.2e10`
- `-2.2`
- `-2.`
- `-.2`
- `-2.3e10`
- `-.2e10`
- `2.2`
- `2.`
- `.2`
- `2.3e10`, `2.3E10`, `2.3e+10`, `2.3e-10`
- `.2e10`

**Comment by jt2594838** [comment]
Good enough, but why not issue this on Jira?

**Comment by LeiRui** [comment]
> Good enough, but how about issuing this on Jira?

Link this pr to [IOTDB-151](https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-151?filter=allopenissues)?

**Comment by jt2594838** [comment]
That should be enough.

**Pull request merged**